File: Strings/remove_duplicates.py
def remove_duplicates(string):
    if len(string) <= 1:
        return string
    
    if string[0] == string[1]:
        return remove_duplicates(string[2:])
    else:
        return string[0] + remove_duplicates(string[2:])
print(remove_duplicates("azxxzy"))

File: Strings/removeDuplicates.java
package Strings;

import java.util.ArrayList;
import java.util.HashMap;

public class removeDuplicates {
    public static void main(String[] args) {
        String str = "geeks";
        int n = str.length();
        HashMap<Character, Integer> map = new HashMap<>();
        StringBuilder sb = new StringBuilder(n);
        for(char c : str.toCharArray()) {
          if(map.containsKey(c))
               continue;
          else {
              map.put(c, 1);
              sb.append(c);
          }
        }
        System.out.println(sb.toString());
    }
}
